test/m_f_c_coin.js

```diff
@@ -18,5 +84,350 @@ contract('MFC_Token', function(accounts) {
           assert.equal(balance_value, 500000000000000000000000000, "500000000000000000000000000 wasn't in the first account");
           assert.equal(totalSupply_value, 500000000000000000000000000, "totalSupply is not 500000000000000000000000000 MFC_Token");
       });
-  });
+    });
 });
+
+contract('MFC_Token tests burn', async function(accounts) {
+
+    /* Task 17 - Create test for MFC_coin burn() with Truffle */
+
+    /* Using Truffle, we check method for MFC_coin burn() and test 4 variants:
+        regular burning - whe*n the sender have enough tokens */
+    it("should transfer 10 tokens to second account and second account should burn 10 tokens", function() {
+        var user_balance;
+        var owner = accounts[0];
+        var user = accounts[1];
+        var token_contract;
+        return token.deployed().then(function(instance) {
+            token_contract = instance;
+            return token_contract.transfer(user, 10, {'from': owner});
+        }).then(function(){
+            return token_contract.balanceOf.call(user);
+        }).then(function(balance) {
+            user_balance_befor = balance.valueOf();
+            return token_contract.burn(10, {'from': user});
+        }).then(function(){
+            return token_contract.balanceOf.call(user);
+        }).then(function(balance) {
+            return balance.toNumber();
+        }).then(function(user_balance_after) {
+            assert.equal(user_balance_befor, user_balance_after + 10, "something went wrong");
```

Comment:
I think it is more logical to check user_balance_befor -10 equals user_balance_after
And user_balance_befor typed with mistake
But, those just for you to know.